sklearn/preprocessing/_data.py

```diff
@@ -834,7 +839,7 @@ def inverse_transform(self, X, copy=None):
             if self.scale_ is not None:
                 inplace_column_scale(X, self.scale_)
         else:
-            X = np.asarray(X)
+            X = asarray(X)
```

Comment:
Instead of implementing a cupy specific version of asarray we should instead have an option to let any NEP-18 compatible implementation go through without calling asarray on it.

For instance:

```python
if not hasattr(X, "__array_function__"):
    X = np.asarray(X)
```

Reply:
Yes, it's definitely better.